virtualaddr.go

```diff
@@ -0,0 +1,172 @@
+package tunnel
+
+import (
```

Comment:
can you clean up this file to have structs at the top and their respective functions grouped?